semantickernel-experimental/src/main/java/com/microsoft/semantickernel/connectors/data/azureaisearch/AzureAISearchVectorStoreRecordCollection.java

```diff
@@ -185,11 +200,12 @@ public Mono<Record> getAsync(
 
         // Use custom mapper if available
         if (mapper != null && mapper.getStorageModelToRecordMapper() != null) {
-            return client.getDocument(key, SearchDocument.class)
+            return searchAsyncClient.getDocument(key, SearchDocument.class)
                 .map(this.options.getVectorStoreRecordMapper()::mapStorageModeltoRecord);
         }
 
-        return client.getDocumentWithResponse(key, this.options.getRecordClass(), selectedFields)
+        return searchAsyncClient
+            .getDocumentWithResponse(key, this.options.getRecordClass(), selectedFields)
```

Comment:
Something to consider. If response.getStatusCode() is anything other than 200, 201, or 202, then something probably went wrong with the request. So it might be better to check if the status is 200, 201, or 202 and then return the value, then check 404 for the "not found", else return SKException "something bad happened". As the code is now, you'll be returning a value even if the response (for example) is a 401 because of bad or expired credentials.

Reply:
Fixed